### ISSUE TITLE
Allow Schema definitions via decorator

### ISSUE THREAD

**Issue by mvanderlee** [vortico/flama#37]
Using type annotations to define the schema messes with IDEs in cases where the schema's post_load returns a different object.

Example:

```python
import marshmallow as ma
import marshmallow.fields as mf

@dataclass
class Foo:
    foo: str
    bar: int


class FooSchema(ma.Schema):
    foo = mf.String()
    bar = mf.Integer(missing=2)

    @ma.post_load
    def make_object(self, data, **kwargs):
        return Foo(**data)
        
@app.route('/', methods=['GET'])
def endpoint(data: FooSchema) -> FooSchema:
    logger.info(f'{type(data)} - {data}')
    return data
```

In this case, data would be and instance of Foo, but the IDE would give type hints for FooSchema.
The same would be true for the response. I'd like to be able to return the instance of Foo, and it be automatically serialized using FooSchema. 
Additionally, marshmallow-dataclass allows us to create a dataclass that has an additional Schema field that we can use to automatically get the schema.

I suggest something like the following:

```python
import logging
from dataclasses import dataclass
from typing import Dict

import marshmallow as ma
import marshmallow.fields as mf
from flama import Flama
from marshmallow_dataclass import dataclass as ma_dataclass

logger = logging.getLogger(__name__)

# Application
app = Flama(
    components=[],      # Without custom components
    title="Foo",        # API title
    version="0.1",      # API version
    description="Bar",  # API description
    schema="/schema/",  # Path to expose OpenAPI schema
    docs="/docs/",      # Path to expose Swagger UI docs
    redoc="/redoc/",    # Path to expose ReDoc docs
)


@ma_dataclass
class Foo2:
    foo: str
    bar: int


@dataclass
class Foo:
    foo: str
    bar: int


class FooSchema(ma.Schema):
    foo = mf.String()
    bar = mf.Integer(missing=2)

    @ma.post_load
    def make_object(self, data, **kwargs):
        return Foo(**data)


# Add decorator to add the schemas to the function, we can then use this during doc generation and data serialization.
def schemas(
    response_schema: ma.Schema = None,
    **request_schemas: Dict[str, ma.Schema]
):
    def decorator(fn):
        fn._request_schemas = request_schemas
        fn._response_schema = response_schema
        return fn

    return decorator


@app.route('/1', methods=['GET'])
def endpoint(data: FooSchema) -> FooSchema:
    logger.info(f'{type(data)} - {data}')
    return data


@app.route('/2', methods=['GET'])
@schemas(data=FooSchema, response_schema=FooSchema)
def endpoint2(data: Foo) -> Foo:
    logger.info(f'{type(data)} - {data}')
    return data


@app.route('/3', methods=['GET'])
def endpoint3(data: Foo2) -> Foo2:
    logger.info(f'{type(data)} - {data}')
    return data
```

**Comment by perdy** [comment]
One of the goal I had in mind when I started to work in this framework was to keep a balance between the complexity of the core codebase and the value it adds to the coder (i.e. increase expresiveness and simplify the coding job).

In my opinion this is a minor gain in front of the major changes we need to add to the user flow and to the injection mechanism. We now force the user to decorate the endpoint and that's opposed to the idea of using the python's typing to keep the app's codebase simple.

Maybe we can consider this kind of behavior as a plugin to enrich the framework features to those users that wanna do the trade off.

**Comment by mvanderlee** [comment]
The decorating is optional, not forced. All it does is allow you to override the schema.
If no decorator is defined, it will use the type hint as before. 

If you still do not want this, I'd at least like to see integration of the marshmallow-dataclass detection that I've added. That would solve most of my use cases.

**Comment by migduroli** [comment]
Closing this issue, as discussed on the PR related to it. We think this is no longer applicable due to the heavy refactoring **Flama** has gone through over the last few months. We really appreciate the effort @mvanderlee. I would suggest you to check the new [documentation](https://flama.dev/), and the new version recently released, `1.3.0`. I think this issue is already resolved. If not, we're happy to re-think about it. Please, let's discuss if beforehand if so, as we're planning more enhancements over the next months, so that we don't duplicate efforts. 
Thanks!
